app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -44,20 +46,21 @@ class FxaRepo(
 ) {
 
     enum class AccountState {
-        // TODO: Are these states accurate?
         // TODO: Later, may need "failed to login": https://github.com/mozilla-mobile/android-components/issues/3712
-        AUTHENTICATED,
-        AUTHENTICATED_NO_PROFILE, // todo: necessary? profile seems to be fetched async so yes?
+        AUTHENTICATED_WITH_PROFILE, // After the profile is fetched async
+        AUTHENTICATED_NO_PROFILE, // Before the profile is fetched async
```

Comment:
Is it possible for an account to never have a profile? Both in terms of 1) profile failed to fetch and 2) account doesn't have a profile (e.g. maybe legacy accounts?). How would we represent this state? Would the account just stay in the AUTHENTICATED_NO_PROFILE state?

Can we document this is some way? e.g. kdoc on each enum to explain the states further.